src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4472,9 +4472,22 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         SkipNonexistentTargets="true"
         RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)">
 
-      <Output TaskParameter="TargetOutputs" ItemName="_AllChildProjectItemsWithTargetPath"/>
+      <Output TaskParameter="TargetOutputs" ItemName="_UnfilteredAllChildProjectItemsWithTargetPath"/>
```

Comment:
Wouldn't it be easier to implement this filtering in the target `GetCopyToOutputDirectoryItems` [after the two CallTarget calls](https://github.com/microsoft/msbuild/pull/4931/files#diff-02872751bc64e0ff785c9b6c1cd919b5R4587), via something like:

```xml
<ItemGroup>
   <_TransitiveItemsToCopyToOutputDirectory Remove="@(_ThisProjectItemsToCopyToOutputDirectory)" />
</ItemGroup>
```

The above removes all the transitive content items that also appear in the current project's content items, thus preventing any transitive content items from overriding the current project's content items.

Another reason for doing it this way is that all the rest of the includes in this target (`_GetCopyToOutputDirectoryItemsFromTransitiveProjectReferences`) can bring transitive content items, so moving it in `GetCopyToOutputDirectoryItems` nicely handles all these items.

Reply:
Moving it makes sense, but I think we would still need the more complicated logic. My understanding is that that above line would filter out duplicate items if they had the same path but I'm looking to filter out items with the same target path, see my test project:

![Screenshot (7)](https://user-images.githubusercontent.com/16010855/71195031-efd2a700-2241-11ea-95a5-0c9f0a5fc205.png)

Is there an easier way to find the overlap of two item groups based on matching metadata than my original implementation?

Reply:
You're right, filtering needs to happen on matching `TargetPath` metadata. Fortunately, doing it in the target `GetCopyToOutputDirectoryItems` should guarantee that all items have `TargetPath` defined.

Unfortunately it gets quite hairy to write the logic. This is the shortest I came up with to remove all `I1` items from `I2` that match on metadata `M`:
```xml
<Project>
    <ItemGroup>
        <I1 Include="a1" M="1"/>
        <I1 Include="b1" M="2"/>
        <I1 Include="c1" M="3"/>
        <I1 Include="d1" M="4"/>

        <I2 Include="a2" M="x"/>
        <I2 Include="b2" M="2"/>
        <I2 Include="c2" M="3"/>
        <I2 Include="d2" M="y"/>
    </ItemGroup>

    <Target Name="Build">
        <ItemGroup>
            <I1_a Include="%(I1.M)" PreviousIdentity="%(I1.Identity)" />
            <I2_a Include="%(I2.M)" PreviousIdentity="%(I2.Identity)" />

            <I2_b Include="@(I2_a)" KeepMetadata="PreviousIdentity"/>

            <I2_a Remove="@(I1_a)" />

            <I2_c Include="@(I2_b)" KeepMetadata="PreviousIdentity"/>
            <I2_c Remove="@(I2_a)" />

            <I2_d Include="%(I2_c.PreviousIdentity)" />

            <I2 Remove="@(I2_d)" />
        </ItemGroup>
        <Message Importance="High" Text="I2: %(I2.Identity) M=[%(I2.M)]" />
    </Target>
</Project>
```

It would be much more useful if we implemented a new msbuild item operation that does all this in one line, something like `<I2 Remove="I1" On="M"/>`. @sfoslund, if I prototype the new operation, would you be willing to finish it (fill in edge cases, make code pretty, tests, shepherd PR, etc)? (cc @livarcocc )

Reply:
@cdmihai  Sure, that's fine with me

Reply:
Awesome. Here's a first draft, that implements this for Remove under targets (but not for remove outside of targets): https://github.com/microsoft/msbuild/pull/4997